xds/internal/xdsclient/xdsresource/unmarshal_lds_test.go

```diff
@@ -1139,7 +1139,7 @@ func (s) TestUnmarshalListener_ServerSide(t *testing.T) {
 				},
 			}),
 			wantName: v3LDSTarget,
-			wantErr:  "is not a terminal filter",
+			wantErr:  "is not a router filter",
```

Comment:
Just to be exhaustive, can we have a test where the last filter *is* a terminal filter, but it is just not the *router* filter.

Reply:
got it @easwars, will update once it's done
thank you